components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -3101,23 +3102,64 @@ private boolean hasPromptContainsConsent(OAuth2Parameters oauth2Params) {
         return prompts != null && Arrays.asList(prompts).contains(OAuthConstants.Prompt.CONSENT);
     }
 
-    private String getUserConsentURL(String sessionDataKey,
-                                     OAuth2Parameters oauth2Params,
-                                     AuthenticatedUser user, OAuthMessage oAuthMessage) throws OAuthSystemException {
-
-        String loggedInUser = user.getAuthenticatedSubjectIdentifier();
-        return EndpointUtil.getUserConsentURL(oauth2Params, loggedInUser, sessionDataKey,
-                OAuth2Util.isOIDCAuthzRequest(oauth2Params.getScopes()), oAuthMessage);
-    }
-
     private String getUserConsentURL(String sessionDataKey,
                                      OAuth2Parameters oauth2Params,
                                      AuthenticatedUser authenticatedUser,
                                      String additionalQueryParams, OAuthMessage oAuthMessage)
             throws OAuthSystemException {
 
-        String userConsentURL = getUserConsentURL(sessionDataKey, oauth2Params, authenticatedUser, oAuthMessage);
-        return FrameworkUtils.appendQueryParamsStringToUrl(userConsentURL, additionalQueryParams);
+        SessionDataCache sessionDataCache = SessionDataCache.getInstance();
+        SessionDataCacheEntry entry;
+        if (oAuthMessage != null) {
+            entry = oAuthMessage.getResultFromLogin();
+        } else {
+            entry = sessionDataCache.getValueFromCache(new SessionDataCacheKey(sessionDataKey));
+        }
+
+        String sessionDataKeyConsent = UUID.randomUUID().toString();
+        String loggedInUser = authenticatedUser.getAuthenticatedSubjectIdentifier();
+        String userConsentURL = EndpointUtil.getUserConsentURL(oauth2Params, loggedInUser,
+                OAuth2Util.isOIDCAuthzRequest(oauth2Params.getScopes()), entry, sessionDataKeyConsent);
+        //Append additional query params to the consent page URL.
+        userConsentURL = FrameworkUtils.appendQueryParamsStringToUrl(userConsentURL, additionalQueryParams);
```

Comment:
can't we combine these two?

Reply:
Yes we can, I  will update the PR

Reply:
Resolved by https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2020/commits/df1132c5c72ed730f0bf91d1091b0d9d5d7ad091